Create F1850SC5-CMIP6 compset

tangq opened the pull request:
Create the F1850SC5-CMIP6 compset (letter "S" in the name stands for special), since the following settings will be different from the F1850C5-CMIP6 compset used by the companion coupled compset (piControl, A_WCYCL1850S_CMIP6)

- l_ieflx_fix flag default to .false.
- Atmosphere and land initial condition data use results at the end (0501-01-01) of the DECK piControl simulation. This is particularly useful to have a better spun-up land state.
- SST and sea ice data use the CMIP6 climatology calculated from 1870-1879.

All the new input files were uploaded to the E3SM input data server: https://web.lcrc.anl.gov/public/e3sm/inputdata/

wlin7 commented:
Hi @tangq , if the intention of this F compset is to use the piControl spun-up cam and clm initial conditions, the changes look all good to me. But do we really want that, esp. for clm initial condition from a coupled run? When land simulation has substantial biases, it would have (probably unintended) consequence on simulations using this F-case.

wlin7 commented:
I see @tangq . My concern is that the land state from coupled simulation may contain larger biases than that from land-only spun-up. Ideally, I think an extended land-only spin up might suit better for general purpose of F-case. That said, this special compset can have its own purpose.

tangq commented:
After discussing with @kaizhangpnl , I found that to turn off IEFLX fixer, ieflx_opt should be 0. l_ieflx_fix = .false. is not effective, and thus removed from the use case xml file.

wlin7 commented:
Do we also want this compset in the master?

PeterCaldwell commented:
I think @wlin7 asks a good question about which finidat file to use. It would be good to use as realistic a finidat as possible because even though our current goal is F2000 vs F1850 sensitivity runs, these F compsets will (hopefully) be what the atmos team uses for all their F runs for v2 and beyond. In this context @thorntonpe describes best practices as spinning up the land using prescribed atmosphere data from reanalysis, using this to run an F case with high-frequency output, then using that output to spin up another prescribed-atmosphere run (see https://acme-climate.atlassian.net/wiki/spaces/Docs/pages/872579110/Adding+support+for+new+atmosphere+resolutions for full description). This 3 step process seems more complex than I expect to be able to convince anyone to do, and it is unclear where we'd get reanalysis for 1850. For our F2010 compset, an alternative (which I prefer) is to use the end of the F20TR simulation. Perhaps the strategy for F1850 should be to start with the F2010 finidat and run it until the land equilibrates, then use that?

As an aside, I just looked in namelist_defaults_clm4_5.xml for what I think are the current F1850 and F2000 finidat files. They start with "climi.I", which makes me think they come from an "I" run = prescribed atmosphere rather than following Peter T's suggestion. Note that 1850_CMIP6_control.xml also sets finidat to to a file that starts with I... which is inconsistent with what I thought @golaz  told me... 

The files from the previous paragraph are: 
clmi.I1850CLM45.ne30_oEC.1155ba0.clm2.r.nc 
clmi.ICRUCLM45.ne30_g16.1155ba0.clm2.r.nc
I1850CLM45.ne30_oECv3.edison.intel.36b43c9.clm2.r.0001-01-06-00000_c20171023.nc 
for 1850,2000, and CMIP6 1850, respectively. I ncdumped them, but they don't have any identifying metadata.

PeterCaldwell commented:
I looked through the changes and they look good to me. My understanding is that the 1850S_cam5_CMIP6.xml file you added differs from 1850_cam5_CMIP6.xml only in adding ieflx_opt=0 and 1850_SCMIP6_control.xml differs only from 1850_CMIP6_control.xml in using a different finidat. Is this true? If so, would it be worth adding comments to these files explaining this difference and saying they were needed because the build infrastructure doesn't allow one to build new compsets by copying an old one then changing a thing or two?

PeterCaldwell commented:
Also, did you try a model run with this new compset yet?

rljacob commented:
I went back and looked at PR #1303 and it looks like l_ieflx_fix was introduced there but not used.  @kaizhangpnl , was it supposed to be used at some point?

If its not used in the code, it should be removed from the namelist and all use cases that try to set it.

rljacob commented:
On further review, PR #1498 was supposed to remove l_ieflx_fix but the code in phys_control.F90 is still there.

golaz commented:
- We should get rid of `l_ieflx_fix` everywhere it appears.  Since both @tangq and I were confused by it (even though I should have remembered it!), others might as well.
- Regarding initial conditions, I'm open to any reasonable (doable) option. The methodology suggested by @thorntonpe is probably the best, but I suspect that nobody would volunteer to apply it. Furthermore it's not obvious how to extend it for 1850. @PeterCaldwell suggestion above seems also reasonable (reuse land initial conditions from the CMIP6 AMIP simulations) and equilibrate it further for 1850. @wlin7, do you think this is a better option?

tangq commented:
@PeterCaldwell , regarding your test run question, I successfully ran this new compset for 1 day on Edison with the new input files on my local directories. I uploaded those new files to the E3SM input data server: https://web.lcrc.anl.gov/public/e3sm/inputdata/, but I don't know the procedures for testing downloading files from the input data server. Can I do such tests before the PR is reviewed and approved? The only instructions I could find are on this page (https://acme-climate.atlassian.net/wiki/spaces/Docs/pages/707002387/E3SM+Input+Data+Servers).

tangq commented:
Once we have a plan for the land initial condition, I can work on it and change the setting in this PR to pick it up.

golaz commented:
@bishtgautam, maybe you want to chime in regarding the best option for land initial condition for the new F1850 compset?

kaizhangpnl commented:
> Sorry if I've asked this question before, but why is l_ieflx_fix still in the model at all if it doesn't actually do anything? Keeping things like this is dangerous, as this PR shows... @kaizhangpnl - can you answer this question?

l_ieflx_fix was removed from the namelist definition file in PR1498 (https://github.com/E3SM-Project/E3SM/pull/1498/files) , but it seems that it was brought back again by https://github.com/E3SM-Project/E3SM/commit/8afcf7d665a707e4178564846788313bb2815edb#diff-c93f649e7d1adfaa25adaf15495ea615. It should have been removed from phys_control.F90 to avoid the confusion.

PeterCaldwell commented:
@kaizhangpnl  - yes, please create a separate PR to remove l_ieflx_fix. Thanks!!!

PeterCaldwell commented:
> @PeterCaldwell , regarding your test run question, I successfully ran this new compset for 1 day on Edison with the new input files on my local directories. I uploaded those new files to the E3SM input data server: https://web.lcrc.anl.gov/public/e3sm/inputdata/, but I don't know the procedures for testing downloading files from the input data server. Can I do such tests before the PR is reviewed and approved? The only instructions I could find are on this page (https://acme-climate.atlassian.net/wiki/spaces/Docs/pages/707002387/E3SM+Input+Data+Servers).

Thanks Qi. My understanding is that so long as the files are on the inputdata server, your branch will download them if they are available. Did this not work?

Overall though, my main question was whether you got this compset to run at all. It sounds like you have, so I'm happy.

PeterCaldwell commented:
Since I haven't heard any objections to my plan of 1). using the end of the transient F-case from the DECK for the F2010 compset and 2). taking that same IC and running it in F1850 mode until it equilibrates, then using the result as the 1850 land IC, I think we should do that (unless @wlin7 or @bishtgautam objects soon!)...

This will require 1). acquiring the clm2.r file for the end of the transient F simulation 2). running it through interpic to get it at ne120 resolution, and 3). using the result as the IC for this compset in order to do a run for long enough for soil moisture to equilibrate. Are these all steps you can do, @tangq ? Or do we need @bishtgautam to do the interpic step?

kaizhangpnl commented:
> @kaizhangpnl - yes, please create a separate PR to remove l_ieflx_fix. Thanks!!!

@PeterCaldwell @tangq @rljacob the use_cases namelist files have the l_ieflx_fix setup. Should they be removed in this separate PR (remove l_ieflx_fix)? 

components//cam/bld/namelist_files/use_cases/1850_cam5_CMIP6.xml:<l_ieflx_fix> .true.</l_ieflx_fix>
components//cam/bld/namelist_files/use_cases/1850_cam5_CMIP6_bgc.xml:<l_ieflx_fix> .true.</l_ieflx_fix>
components//cam/bld/namelist_files/use_cases/1950_cam5_CMIP6-HR.xml:<l_ieflx_fix> .true.</l_ieflx_fix>
components//cam/bld/namelist_files/use_cases/1950_cam5_CMIP6-LR.xml:<l_ieflx_fix> .true.</l_ieflx_fix>
components//cam/bld/namelist_files/use_cases/1950_cam5_CMIP6-LRtunedHR.xml:<l_ieflx_fix> .true.</l_ieflx_fix>
components//cam/bld/namelist_files/use_cases/20TR_cam5_CMIP6.xml:<l_ieflx_fix> .true.</l_ieflx_fix>
components//cam/bld/namelist_files/use_cases/20TR_cam5_CMIP6_bgc.xml:<l_ieflx_fix> .true.</l_ieflx_fix>

tangq commented:
@PeterCaldwell , we don't need to interpolate the land IC file to ne120 grids, because this is for the ne30 compset. I can do all steps (including interpic).

wlin7 commented:
The method, proposed by @PeterCaldwell and reiterated by @golaz , sounds good to me for generating finidat for F1850 if F1850 can run with the clm2.r from the end of F20TR  simulation. I am just not sure whether differences in landuse for 1850 and PD could fail some consistency check during initialization (e.g., urban grids).

wlin7 commented:
I was over concerned about using finidat from F20TR run. Just did a 1-day test run using this compset and cam.i and clm2.r of 2010-01-01 from the DECKv1b_H1 simulation.  It went well.

One more issue regarding the path to prescribed sst_ice file. The current path is inputdata/atm/cam/sst. The file  sst_ice_CMIP6_DECK_E3SM_1x1_1850_clim_c190125.nc was not available on NERSC. The inputdata server has the file under inputdata/ocn/docn7/SSTDATA, along with some other sst_ice_CMIP6_DECK_E3SM files. For the time being, I copied the 1850 file to the default path in this compset. We do need to resolve this inconsistency, one way or another.

tangq commented:
> I was over concerned about using finidat from F20TR run. Just did a 1-day test run using this compset and cam.i and clm2.r of 2010-01-01 from the DECKv1b_H1 simulation. It went well.
> 
> One more issue regarding the path to prescribed sst_ice file. The current path is inputdata/atm/cam/sst. The file sst_ice_CMIP6_DECK_E3SM_1x1_1850_clim_c190125.nc was not available on NERSC. The inputdata server has the file under inputdata/ocn/docn7/SSTDATA, along with some other sst_ice_CMIP6_DECK_E3SM files. For the time being, I copied the 1850 file to the default path in this compset. We do need to resolve this inconsistency, one way or another.

@wlin7 , I was not sure which input directory is the current default, so I used the same one as the F1850C5-CMIP6 compset. I will change it to inputdata/atm/cam/sst and move the sst & sea ice file on the input data server as well.

tangq commented:
It seems that we all agree with the method @PeterCaldwell proposed to generate the land IC. I will work on it. How long is a good length for equilibrium? 50 years?

rljacob commented:
Which namelist has the path to the sst file?  I assume its docn so the file should go in inputdata/ocn/docn7/SSTDATA

wlin7 commented:
@rljacob , it is ice_in and docn.streams.txt.prescribed. The file is used by both docn and cice. Historically, such sst/ice files required by F-cases are put on  inputdata/atm/cam/sst. Starting to see some compsets assuming the the path of inputdata/ocn/docn7/SSTDATA.

tangq commented:
> Which namelist has the path to the sst file? I assume its docn so the file should go in inputdata/ocn/docn7/SSTDATA

The sst file path is the file ice_in and docn.streams.txt.prescribed and controlled by the setting in cime/src/components/data_comps/docn/cime_config/config_component.xml. What the latest default path for those files? I will make change to this PR and consistently on the input data server.

rljacob commented:
Lets start using inputdata/ocn/docn7/SSTDATA.   The cam one is from when CAM had a true standalone mode an its own data ocean model.  Those don't exist anymore.

PeterCaldwell commented:
> It seems that we all agree with the method @PeterCaldwell proposed to generate the land IC. I will work on it. How long is a good length for equilibrium? 50 years?

I think the strategy should be to run for ~20 yrs, look at timeseries some measure of land spin-up, and continue running until that timeseries stabilizes. @bishtgautam and @thorntonpe  - what variable(s) do you think would be appropriate to use to measure land spin up?

tangq commented:
> Lets start using inputdata/ocn/docn7/SSTDATA. The cam one is from when CAM had a true standalone mode an its own data ocean model. Those don't exist anymore.

I modified the path to use inputdata/ocn/docn7/SSTDATA. The SST and sea ice file was also moved to that directory on the input data server.

tangq commented:
> > It seems that we all agree with the method @PeterCaldwell proposed to generate the land IC. I will work on it. How long is a good length for equilibrium? 50 years?
> 
> I think the strategy should be to run for ~20 yrs, look at timeseries some measure of land spin-up, and continue running until that timeseries stabilizes. @bishtgautam and @thorntonpe - what variable(s) do you think would be appropriate to use to measure land spin up?

I retrieved the land restart file at 2015-01-01 from the DECK A1 simulation and configured a run to use this F1850SC5-CMIP6 compset with that land restart file. I will do one submission. It can finish about 18 years. I will post the link to the results here when it's done.

PeterCaldwell commented:
> > > It seems that we all agree with the method @PeterCaldwell proposed to generate the land IC. I will work on it. How long is a good length for equilibrium? 50 years?
> > 
> > 
> > I think the strategy should be to run for ~20 yrs, look at timeseries some measure of land spin-up, and continue running until that timeseries stabilizes. @bishtgautam and @thorntonpe - what variable(s) do you think would be appropriate to use to measure land spin up?
> 
> I retrieved the land restart file at 2015-01-01 from the DECK A1 simulation and configured a run to use this F1850SC5-CMIP6 compset with that land restart file. I will do one submission. It can finish about 18 years. I will post the link to the results here when it's done.

That sounds great. Thanks!

tangq commented:
The F1850SC5-CMIP6 land IC simulation will finish 18 years today. The results are on Edison:/global/cscratch1/sd/tang30/ACME_simulations/edison.20190125_maint-1.0.F1850SCMIP6_landIC.ne30_oECv3/run. Next, we will need to know the state of land spin-up to decide if it is necessary to extend this run.

tangq commented:
@PeterCaldwell and I discussed options about the land IC, we decided to use the same land IC file as the transient compset (F20TRC5-CMIP6) to maintain consistency between F1850 and AMIP simulations. I have made the change on this PR.

wlin7 commented:
Hi @kaizhangpnl , @PeterCaldwell , @golaz , 
Can you please take a look of this PR? Qi requested you to give a review. I have also reviewed and performed tests with this PR. Thanks.

rljacob commented:
Does this need to be on master as well as maint-1.0?

tangq commented:
Eventually we want this on master too. My understanding is that some new files will be needed to make this compset compatible with master, but I don't know how to create those files. So I only added it to maint-1.0.

wlin7 commented:
@tangq , do you want all your suggested reviewers to approve for this PR? I feel it is safe to merge.

tangq commented:
@wlin7 , I think so too. It should be save to merge.

PeterCaldwell commented:
Sorry for being tardy in my review. Feel free to merge without my review. If you want to wait, I'll try to find time tomorrow.

wlin7 commented:
No problem at all, @PeterCaldwell . It is definitely very helpful if you can find time to take a look. There is no hurry for this PR.

wlin7 commented:
Thanks @PeterCaldwell . I am going to merge it today.

wlin7 commented:
Merged to maint-1.0.
